### ISSUE TITLE
[Bug]: Nothing works due to Cloudflare check

### ISSUE THREAD

**Issue by alpatron** [FurryCoders/FAAPI#10]
### Version

3.11.3

### What happened?

Hello, today I tried to use FAAPI and discovered that it stopped working.

Any request I make upon FAAPI ends with a `requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: [...]`[^2] exception. Looking into the response gotten from the server, I can see that the content of the response starts with `<!DOCTYPE html><html lang="en-US"><head><title>Just a moment...</title>`, which corresponds to a [Cloudflare challenge](https://developers.cloudflare.com/fundamentals/get-started/concepts/cloudflare-challenges/).

I don't know whether this is because FA changed something in their hosting to require all visitors to pass the challenge[^1], or whether FA is singling my connection for these checks, but in either case, I do think that FAAPI should at least throw a better error, so the user can see that a Cloudlfare check is preventing FAAPI from working, because as of now, the error thrown is a simple 403, which can mean many things.

[^1]: It seems so, because testing on my two devices, I always get the challenge when trying to access FA.
[^2]: `[...]` in excerpt signifies omisison.

### How to reproduce the bug?

Try to do anything that sends a request to FA.

**Comment by MatteoCampinoti94** [comment]
Thank for the suggestion of a CloudFlare-specific exception, I will add it.

**Comment by Deer-Spangle** [comment]
Yup, FA was under cloudflare anti-bot protection for about 12 hours today (and 8 hours on a couple previous days)
I've spotted sharp spikes in the number of online users before these blackouts, so I suspect they're getting hit by some DDoS attacks.

It should be available again now though

**Comment by MatteoCampinoti94** [comment]
Thanks for the update @Deer-Spangle :)
